File: sprint5/By languages/go/lesson7code2.go
func smallLeftRotation(a *Node) *Node {
  // Задаём обозначения.
  b := a.right
  C := b.left

  // Переусыновляем вершины.
  a.right = C
  b.left = a

  // Корректируем высоты в зависимости от того, равны ли высоты C и R.
  if C.height == a.height {
    a.height -= 1
    b.height += 1
  } else {
    a.height -= 2
  }
  return b
}

File: sprint5/By languages/JS/lesson7code2.js
function small_left_rotation(a) {
  // Задаём обозначения.
  const b = a.right;
  const C = b.left;

  // Переусыновляем вершины.
  a.right = C;
  b.left = a;

  // Корректируем высоты в зависимости от того, равны ли высоты C и R.
  if (C.height === a.height) {
    a.height -= 1;
    b.height += 1;
  } else {
    a.height -= 2;
  }
  return b;
}
